library/kani/src/contracts.rs

```diff
@@ -225,4 +225,27 @@
 //! Rust pointer type (`&T`, `&mut T`, `*const T` or `*mut T`). In addition `T`
 //! must implement [`Arbitrary`](super::Arbitrary). This is used to assign
 //! `kani::any()` to the location when the function is used in a `stub_verified`.
+//!
+//! ## History Variables
+//!
+//! Additionally, an ensures clause is allowed to refer to history computations
+//! via an `old` monad. Any instance of `old(computation)` will evaluate the
+//! computation before the function is called. It is required that this computation
+//! is effect free and closed with respect to the function arguments.
```

Comment:
A very important point here is to mention whether the absence of side effects here is _required_ or whether it is _checked_. I believe the latter is the case, because CBMC will deny any side-effect use in contracts. 

Might even warrant a test case.

Reply:
Required and not enforced by kani. Added a new test case in that fails due to side effects within the old computation. You may still be right for other kani contracts, but it isn't checked for old.